### PR TITLE
[release-v2.7] OOB Release for Longhorn 1.4.0

### PR DESCRIPTION
**Pull request by pennyscissors**
Issue: https://github.com/rancher/rancher/issues/39750

### CODE REVIEW

**Comment by pennyscissors** [comment]
## Charts Checklist (built for v0.3.x charts-build-scripts)

### Checkpoint 0: Validate `release.yaml`

Validation steps:
- [x] Each chart version in `release.yaml` DOES NOT modify an already released chart. If so, stop and modify the versions so that it releases a net-new chart.
- [x] Each chart version in `release.yaml` IS exactly 1 more patch version than the last released chart version. If not, stop and modify the versions so that it releases a net-new chart.

### Checkpoint 1: Compare contents of assets/ to charts/

Validation steps:
- [x] Running `make unzip` to regenerate the `charts/` from scratch, then `git diff` to check differences between `assets/` and `charts/` yields NO differences or innocuous differences.

IMPORTANT: Do not undo these changes for future steps since we want to keep the charts/ that match the current contents of assets!

### Checkpoint 2: Compare assets against index.yaml

Validation steps:
- [x] The `index.yaml` file has an entry for each chart version.
- [x] The `index.yaml` entries for each chart matches the `Chart.yaml` for each chart.
- [x] Each chart has ALL required annotations
  - kube-version annotation
  - rancher-version annotation
  - permits-os annotation (indicates Windows and/or Linux)

### Checkpoint 3: Logical Checks

Validation steps (requires manual inspection of differences between previous and latest chart version):
- [x] Ensure no RC images are included.
- [x] Ensure all mirrored images include the right prefix.
- [x] Ensure any new images have `repository` and `tag` on the same level to ensure the rancher-images.txt is generated correctly for it for airgap scenarios.

**Comment by pennyscissors** [comment]
Validation shows the following warning, but this is a known issue for the Longhorn team:
```
Checking annotations on chart charts/longhorn/101.2.0+up1.4.0
WARN: Chart.yaml missing 'catalog.cattle.io/permits-os'
```

File: packages/longhorn-crd/longhorn-1.4/package.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+url: https://github.com/longhorn/charts.git
+subdirectory: charts/longhorn
+commit: 82be2970c8ffb8b50d6fcad77a3326d8e2434012
+version: 101.2.0
```
Comment (eliyamlevy) [inline packages/longhorn-crd/longhorn-1.4/package.yaml:4]: This needs to be changed to `v102.0.0` to match the upcoming release.

Comment (PhanLe1010) [reply packages/longhorn-crd/longhorn-1.4/package.yaml:4]: @eliyamlevy  This change will be shipped by another chart PR https://github.com/rancher/charts/pull/2385/files#diff-5050e5069d5e0ff8413a60d784e47c9299f745c1c0c58d410c0ca853b6432787R4 (come out together with Rancher v2.7.2)

Comment (innobead) [reply packages/longhorn-crd/longhorn-1.4/package.yaml:4]: Does this mean Longhorn 1.4.0 actually not released yet? But will release together with 2.7.2?

cc @rebeccazzzz

Comment (pennyscissors) [reply packages/longhorn-crd/longhorn-1.4/package.yaml:4]: @eliyamlevy This is for an OOB release, not 2.7.2. [This other PR](https://github.com/rancher/charts/pull/2385) is handling the 102.x.x versioning for the Longhorn chart that will be released with or after 2.7.2.

**Comment by innobead** [comment]
> Validation shows the following warning, but this is a known issue for the Longhorn team:
> 
> ```
> 
> Checking annotations on chart charts/longhorn/101.2.0+up1.4.0
> 
> WARN: Chart.yaml missing 'catalog.cattle.io/permits-os'
> 
> ```

@PennyScissors Is this annotation required and for what reason? @PhanLe1010 @weizhe0422 did u know the context?

**Comment by pennyscissors** [comment]
@innobead That annotation is used to allow deploying the chart in a hybrid cluster with windows and linux nodes, but it requires specific node-selectors and tolerations. The annotation was purposely removed initially as a temporary measure to prevent an issue from happening, but it should've been added upstream after the issue was fixed upstream. Please read the following for more context:
https://github.com/rancher/rancher/issues/38329#issuecomment-1189348884
https://github.com/rancher/rancher/issues/37030

**Review by aiyengar2 — APPROVED** [review]

**Pull request merged**